packages/tic-tac-toe-bot/src/bot.ts

```diff
@@ -0,0 +1,184 @@
+import { ethers } from "ethers";
+import { v4 } from "uuid";
+
+const bn = ethers.utils.bigNumberify;
+const ZERO = bn(0);
+
+function checkDraw(board) {
+  return board.every(row => row.every(square => !bn(square).eq(ZERO)));
+}
+
+function checkVictory(board, player) {
+  return (
+    checkHorizontalVictory(board, player) ||
+    checkVerticalVictory(board, player) ||
+    checkDiagonalVictory(board, player) ||
+    checkCrossDiagonalVictory(board, player)
+  );
+}
+
+function checkHorizontalVictory(board, player) {
+  let idx;
+  const victory = board.some((row, index) => {
+    idx = index;
+    return row.every(square => bn(square).eq(bn(player)));
+  });
+
+  if (victory) {
+    return {
+      idx,
+      winClaimType: 0
+    };
+  }
+  return false;
+}
+
+function checkVerticalVictory(board, player) {
+  let idx;
+  const victory = board[0].some((columnStart, index) => {
+    idx = index;
+    return (
+      bn(columnStart).eq(bn(player)) &&
+      bn(board[1][index]).eq(bn(player)) &&
+      bn(board[2][index]).eq(bn(player))
+    );
+  });
+
+  if (victory) {
+    return {
+      idx,
+      winClaimType: 1
+    };
+  }
+  return false;
```

Comment:
Can single line this `return victory ? { idx, winClaimType: 1 } : false`.

Also separately, it seems error prone to return `bool | { idx: number, winClaimType: number }`. Should return a consistent type.